plugins/outputs/health/health.go

```diff
@@ -153,14 +164,15 @@ func (h *Health) ServeHTTP(rw http.ResponseWriter, _ *http.Request) {
 
 // Write runs all checks over the metric batch and adjust health state.
 func (h *Health) Write(metrics []telegraf.Metric) error {
+	h.firstMetricReceived = true
+	h.lastMetricTime = time.Now()
```

Comment:
How about initializing the `lastMetricTime` in `Connect`?

Reply:
This is because `Connect` is guaranteed to be called before the first `Write`, correct?

Reply:
I left the initialization in `NewHealth` (which is called from `init`). If `Connect` is preferred, I can move it there.

Reply:
Yes, `Connect` would be better. The `init` function is called when importing this plugin so you might get timeouts if setting up Telegraf takes long. `Connect` is called _after_ `Init` but before the first `Write` and is the better place I think.

Reply:
With the latest change, `lastMetricTime` is initialized as 0 and it is only updated in `Write`.  Whe can add an initialization to `now` on `Connect`. This will help catch cases where telegraf starts but no metric is written before the max time elapses.
The current behaviour waits for the first metric to be written to start checking if the time between metrics is longer than expected. This was by design, to prevent the health output to fail during startup. However, if `Connect` is called 'close enough' to when everything should already be working, we can set `lastMetricTime = now` in connect also.

Reply:
with the last change, `lastMetricTime` is set to `now` in `Connect`.